app/src/main/java/practicum/getfitla_v3/Exercise.java

```diff
@@ -5,58 +5,94 @@
         setContentView(R.layout.exercise);
-        mContext = this;
-        recyclerView = (RecyclerView) findViewById(R.id.reyclerView);
-        recyclerView.setHasFixedSize(true);
-        recyclerView.setLayoutManager(new LinearLayoutManager(this));
-        exerciseList = new ArrayList<>();
-        exerciseList.add(
-                new ExerciseItemFormat(
-                        1,
-                        "Apple MacBook Air Core i5 5th Gen - (8 GB/128 GB SSD/Mac OS Sierra)",
-                        "13.3 inch, Silver, 1.35 kg",
-                        4.3,
-                        60000,
-                        R.drawable.building));
-
-        exerciseList.add(
-                new ExerciseItemFormat(
-                        1,
-                        "Dell Inspiron 7000 Core i5 7th Gen - (8 GB/1 TB HDD/Windows 10 Home)",
-                        "14 inch, Gray, 1.659 kg",
-                        4.3,
-                        60000,
-                        R.drawable.building));
-
-        exerciseList.add(
-                new ExerciseItemFormat(
-                        1,
-                        "Microsoft Surface Pro 4 Core m3 6th Gen - (4 GB/128 GB SSD/Windows 10)",
-                        "13.3 inch, Silver, 1.35 kg",
-                        4.3,
-                        60000,
-                        R.drawable.building));
-
-        mAdapter = new ExerciseListAdapter(mContext, exerciseList);
-        recyclerView.setAdapter(mAdapter);
-
-        mAdapter.setClickListener(this);
+        final Exercise this_exercise = this;
+        new DownloadWebpageTask(new AsyncResult() {  //This essentially begins the whole database processing fiasco.
+            @Override
+            public void onResult(JSONObject object) {
+                processJson(object);
+                recyclerView = findViewById(R.id.reyclerView);
+                recyclerView.setHasFixedSize(true); //sets a fixed size for the recycler view size, not the elements in the recycler view
+                recyclerView.setLayoutManager(new LinearLayoutManager(this_exercise)); //setting the orientation of the recyclerview (by default it is vertical
+
+                //each item in the arraylist will be an object. Each object is created in the detail file and add()ed here
+                //this will become the google sheets feed soon
+                ExerciseListAdapter adaptor = new ExerciseListAdapter(this_exercise, exerciseList);
+                recyclerView.setAdapter(adaptor);
+                recyclerView.setHasFixedSize(true);
+                recyclerView.setLayoutManager(new LinearLayoutManager(this_exercise));
+                mAdapter = new ExerciseListAdapter(this_exercise, exerciseList);
+                recyclerView.setAdapter(mAdapter);
+                mAdapter.setClickListener(this_exercise);
+            }
+        }).execute("https://spreadsheets.google.com/tq?key=1jFTMl8k53itUpU2NAXjAIBbdEChwcVJ3N-b4mQYi4qc");
```

Comment:
Also, if this is a static URL, consider storing it in a global or final variable.